django/template/base.py

```diff
@@ -227,6 +227,7 @@ def compile_nodelist(self):
             tokens, self.engine.template_libraries, self.engine.template_builtins,
             self.origin,
         )
+        parser.template_name = self.origin.template_name
```

Comment:
Curious, is there a reason to not use `parser.origin.template_name` in `do_extends` instead of setting this as a separate attribute?